test/parallel/sdk.API.claimRewards.test.ts

```diff
@@ -0,0 +1,450 @@
+/*
+ *
+ * @group sdk
+ */
+import { jest } from "@jest/globals";
+import { Keyring } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { BN } from "@polkadot/util";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { User } from "../../utils/User";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { Xyk } from "../../utils/xyk";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import {
+  claimRewardsAll,
+  getLiquidityAssetId,
+  getRewardsInfo,
+  promotePool,
+} from "../../utils/tx";
+import {
+  getEventResultFromMangataTx,
+  getNextAssetId,
+} from "../../utils/txHandler";
+import { ExtrinsicResult, waitForRewards } from "../../utils/eventListeners";
+import { BN_ZERO } from "@mangata-finance/sdk";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser: User;
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let soloTokenId: BN;
+let poolTokenIds: BN[];
+let liqIds: BN[];
+let tokenAmounts: BN[];
+const defaultCurrencyValue = new BN(250000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  await setupApi();
+  await setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintNative(sudo)
+  );
+
+  [testUser] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
```

Comment:
You can create the user at line 57 and mint the tokens in the batch at line 58, so we save one block test execution

Reply:
But we need to receive info about testUser, I think it would be better if we get it as a constant right away, wouldn't it?

Reply:
what I mean is that you can create the user and mint the tokens in the sudoBatch above. am I right?

Reply:
Yes, but after that, I need to use testUsers' info in the tests below, for I need to extract them from the results of the batch function l and I do not see the point in this

Reply:
I mean this: 
![image](https://github.com/mangata-finance/mangata-e2e/assets/15139384/fee33624-4542-4821-9bba-1b07fab56fa9)